src/Microsoft.Net.Http.Headers/ContentDispositionHeaderValue.cs

```diff
@@ -69,13 +69,14 @@ public IList<NameValueHeaderValue> Parameters
 
         public StringSegment Name
         {
-            get { return GetName(NameString); }
+            get { return HeaderUtilities.RemoveQuotes(GetName(NameString)); }
```

Comment:
This is also turning the StringSegment into a string and back into a StringSegment negating the benefits. GetName should take a StringSegment.